### PR TITLE
Fix misspelled cmake variable - build all tests

### PR DESCRIPTION
**Pull request by crtrott**
Apparently we were not building all things since mid July or so https://github.com/kokkos/kokkos/pull/7106

### CODE REVIEW

**Review by ldh4 — APPROVED** [review]

**Review by diehlpk — APPROVED** [review]
LGTM!

**Comment by crtrott** [comment]
@ndellingwood can you retest the Intel compiler issues we tried to fix, because maybe it only worked because we didn't compile stuff ...

**Comment by Rombur** [comment]
Looks like we broke `view_alloc` and `view_alloc_int`. These tests are failing with serial and with OpenMP. OpenMPTarget and OpenACC have tests that are not compiling.

**Comment by dalg24** [comment]
cc @seyonglee @rgayatri23

**Comment by ndellingwood** [comment]
@crtrott ouch... I was able to successfully compile this PR using intel/2021.3 icpc (checked the serial backend) with the reproducer from here https://github.com/kokkos/kokkos/issues/7084#issue-2363358149:

```
cmake -DCMAKE_CXX_COMPILER=icpc -DCMAKE_CXX_FLAGS="-no-ip" -DKokkos_ENABLE_SERIAL=ON -DKokkos_ARCH_SKX=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_BUILD_TYPE=RELEASE -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF $KOKKOS_PATH
```

These are the tests that compiled:
```
[ndellin@kokkos-dev-2 SemsIntel2021_3]$ ctest -N
Test project /home/ndellin/kokkos/testing/SemsIntel2021_3
  Test  #1: Kokkos_CoreUnitTest_Serial1
  Test  #2: Kokkos_CoreUnitTest_Serial2
  Test  #3: Kokkos_CoreUnitTest_Default
  Test  #4: Kokkos_CoreUnitTest_LegionInitialization
  Test  #5: Kokkos_CoreUnitTest_PushFinalizeHook
  Test  #6: Kokkos_CoreUnitTest_ScopeGuard
  Test  #7: Kokkos_CoreUnitTest_Develop
  Test  #8: Kokkos_CoreUnitTest_PushFinalizeHookTerminateRegex
  Test  #9: Kokkos_CoreUnitTest_PushFinalizeHookTerminateFails
  Test #10: Kokkos_CoreUnitTest_KokkosP
  Test #11: Kokkos_CoreUnitTest_ToolIndependence
  Test #12: Kokkos_ProfilingTestLibraryLoadHelp
  Test #13: Kokkos_ProfilingTestLibraryCmdLineHelp
  Test #14: Kokkos_ProfilingTestLibraryLoad
  Test #15: Kokkos_ProfilingTestLibraryCmdLine
  Test #16: Kokkos_CoreUnitTest_StackTraceTest
  Test #17: Kokkos_IncrementalTest_SERIAL
  Test #18: Kokkos_CoreUnitTest_CTestDevice
  Test #19: Kokkos_CoreUnitTest_CMakePassCmdLineArgs0
  Test #20: Kokkos_CoreUnitTest_DeviceAndThreads
  Test #21: Kokkos_ContainersUnitTest_Serial
  Test #22: Kokkos_UnitTest_Sort
  Test #23: Kokkos_UnitTest_Random
  Test #24: Kokkos_AlgorithmsUnitTest_StdSet_A
  Test #25: Kokkos_AlgorithmsUnitTest_StdSet_B
  Test #26: Kokkos_AlgorithmsUnitTest_StdSet_C
  Test #27: Kokkos_AlgorithmsUnitTest_StdSet_D
  Test #28: Kokkos_AlgorithmsUnitTest_StdSet_E
  Test #29: Kokkos_AlgorithmsUnitTest_StdSet_Team_A
  Test #30: Kokkos_AlgorithmsUnitTest_StdSet_Team_B
  Test #31: Kokkos_AlgorithmsUnitTest_StdSet_Team_C
  Test #32: Kokkos_AlgorithmsUnitTest_StdSet_Team_D
  Test #33: Kokkos_AlgorithmsUnitTest_StdSet_Team_E
  Test #34: Kokkos_AlgorithmsUnitTest_StdSet_Team_F
  Test #35: Kokkos_AlgorithmsUnitTest_StdSet_Team_G
  Test #36: Kokkos_AlgorithmsUnitTest_StdSet_Team_H
  Test #37: Kokkos_AlgorithmsUnitTest_StdSet_Team_I
  Test #38: Kokkos_AlgorithmsUnitTest_StdSet_Team_L
  Test #39: Kokkos_AlgorithmsUnitTest_StdSet_Team_M
  Test #40: Kokkos_AlgorithmsUnitTest_StdSet_Team_P
  Test #41: Kokkos_AlgorithmsUnitTest_StdSet_Team_Q
  Test #42: Kokkos_UnitTest_SIMD

Total Tests: 42
```

**Comment by ndellingwood** [comment]
@crtrott two subtest failures, already mentioned:

```
[ RUN      ] serial.view_alloc
Test failure: Didn't encounter wanted events
BeginFence { "Kokkos::View::initialization", 1,6}
EndFence { 6}
/home/ndellin/kokkos/core/unit_test/TestWithoutInitializing.hpp:140: Failure
Value of: success
  Actual: false
Expected: true
[  FAILED  ] serial.view_alloc (0 ms)
[ RUN      ] serial.view_alloc_exec_space
[       OK ] serial.view_alloc_exec_space (0 ms)
[ RUN      ] serial.view_alloc_int
Test failure: Didn't encounter wanted events
BeginFence { "HostSpace fence", 1,8}
EndFence { 8}
BeginFence { "Kokkos::View::initialization via memset", 1,9}
EndFence { 9}
/home/ndellin/kokkos/core/unit_test/TestWithoutInitializing.hpp:184: Failure
Value of: success
  Actual: false
Expected: true
[  FAILED  ] serial.view_alloc_int (0 ms)
```

**Comment by Rombur** [comment]
Only `view_alloc` and `view_alloc_int` are failing for HIP.

**Comment by crtrott** [comment]
Pushed a fix: the fence names changed for view_alloc.

**Review by rgayatri23 — APPROVED** [review]

**Comment by dalg24** [comment]
```
In file included from /var/jenkins/workspace/Kokkos_PR-7238/build/core/unit_test/openmptarget/TestOpenMPTarget_ViewBadAlloc.cpp:2:
/var/jenkins/workspace/Kokkos_PR-7238/core/unit_test/TestViewBadAlloc.hpp:78:35: error: no member named 'SharedSpace' in namespace 'Kokkos'
   78 |       test_view_bad_alloc<Kokkos::SharedSpace>();
      |                           ~~~~~~~~^
/var/jenkins/workspace/Kokkos_PR-7238/core/unit_test/TestViewBadAlloc.hpp:81:35: error: no member named 'SharedHostPinnedSpace' in namespace 'Kokkos'
   81 |       test_view_bad_alloc<Kokkos::SharedHostPinnedSpace>();
      |                           ~~~~~~~~^
```

**Comment by seyonglee** [comment]
> ```
> In file included from /var/jenkins/workspace/Kokkos_PR-7238/build/core/unit_test/openmptarget/TestOpenMPTarget_ViewBadAlloc.cpp:2:
> /var/jenkins/workspace/Kokkos_PR-7238/core/unit_test/TestViewBadAlloc.hpp:78:35: error: no member named 'SharedSpace' in namespace 'Kokkos'
>    78 |       test_view_bad_alloc<Kokkos::SharedSpace>();
>       |                           ~~~~~~~~^
> /var/jenkins/workspace/Kokkos_PR-7238/core/unit_test/TestViewBadAlloc.hpp:81:35: error: no member named 'SharedHostPinnedSpace' in namespace 'Kokkos'
>    81 |       test_view_bad_alloc<Kokkos::SharedHostPinnedSpace>();
>       |                           ~~~~~~~~^
> ```

The OpenACC backend has the same errors since `Kokkos::SharedSpace` and Kokkos::SharedHostPinnedSpace` are not defined in the OpenACC backend (and OpenMPTarget backend too).

**Comment by dalg24** [comment]
Interesting, the tests are guarded with `is constexpr (Kokkos::has_shared[_host_pinned]_space)`
https://github.com/kokkos/kokkos/blob/f5c527d7cb12e572643e0a0810ba32fcb6d8ffcc/core/unit_test/TestViewBadAlloc.hpp#L76-L83

**Review by nmm0 — APPROVED** [review]
Wow

File: core/unit_test/TestViewBadAlloc.hpp [hunk 1]
```diff
@@ -68,19 +68,17 @@ TEST(TEST_CATEGORY, view_bad_alloc) {
...
   test_view_bad_alloc<MemorySpace>();
 
+#if !defined(KOKKOS_ENABLE_OPENACC) && !defined(KOKKOS_ENABLE_OPENMPTARGET)
   constexpr bool execution_space_is_device =
       std::is_same_v<ExecutionSpace, Kokkos::DefaultExecutionSpace> &&
       !std::is_same_v<Kokkos::DefaultExecutionSpace,
                       Kokkos::DefaultHostExecutionSpace>;
 
   if constexpr (execution_space_is_device) {
-    if constexpr (Kokkos::has_shared_space) {
```
Comment (nmm0) [inline core/unit_test/TestViewBadAlloc.hpp:77]: Is this a compiler bug or do we not properly set `has_shared_space`?

**Review by masterleinad — DISMISSED** [review]

File: core/unit_test/TestViewBadAlloc.hpp [hunk 2]
```diff
@@ -68,19 +68,17 @@ TEST(TEST_CATEGORY, view_bad_alloc) {
 
   test_view_bad_alloc<MemorySpace>();
 
+#if !defined(KOKKOS_ENABLE_OPENACC) && !defined(KOKKOS_ENABLE_OPENMPTARGET)
```
[the remark this answers is not in the document]
Comment (dalg24) [reply core/unit_test/TestViewBadAlloc.hpp:71]: What Daniel said

[the remark this answers is not in the document]
Comment (crtrott) [reply core/unit_test/TestViewBadAlloc.hpp:71]: overlooked that we had both a constexpr variable and a macro

**Comment by dalg24** [comment]
I suppose there is plenty of approvals already

**Comment by masterleinad** [comment]
```
/var/jenkins/workspace/Kokkos_PR-7238/core/src/View/Kokkos_ViewAlloc.hpp(65): error #20014-D: calling a __host__ function from a __host__ __device__ function is not allowed
```

**Comment by dalg24** [comment]
Ignoring, clearly unrelated

**Pull request merged**